### ISSUE TITLE
Feature Request - implement a queue or generator to handle test suites 

### ISSUE THREAD

**Issue by llaville** [sebastianbergmann/phpunit#1921]
PHPUnit suffer from its legacy architecture, because we have to wait final tests count, before starting at least first of them.

In my case, I've one hundred of *Test.php files that produced about 4800 unit tests (using data provider).
When I run full tests suite, I've to wait a long time before to see any dot in stdout (default printer results), time while PHPUnit count all tests, then after results come quickly.

My suggestion will be to implement a generator solution that will proceed to each *Test.php file ASAP !
Hope my explains are enough.

**Comment by llaville** [comment]
I've found an alternative with a phing task.
See https://github.com/llaville/php-compatinfo-db/commit/23230c749bc92efb29a15a7137d06fb08280a9ea

CAUTION: the phar distrib used with `phunit.pharlocation` phing property should have `require_once`instead of `require` (see #1924)

**Comment by sebastianbergmann** [comment]
Duplicate of #10.

**Comment by llaville** [comment]
Just to let you know that I've implemented in Phing PHPUnitTask an alternative solution based on PHP 5.5 generator feature. It will be proposed next week. See http://www.phing.info/trac/ticket/1222

In waiting for native PHPUnit solution with #10
